timestamp-impl/src/test/java/com/palantir/timestamp/PersistentUpperLimitTest.java

```diff
@@ -78,6 +97,19 @@ public void shouldPersistAnIncreasedTimestamp() {
     }
 
     @Test
+    public void shouldNotChangeTheCurrentUpperLimitIfItFailsToPersist() {
+        doThrow(RuntimeException.class).when(boundStore).storeUpperLimit(anyLong());
+
+        try {
+            upperLimit.increaseToAtLeast(INITIAL_UPPER_LIMIT + 10);
+        } catch (Exception e) {
```

Comment:
There's a separate test checking how we deal with exceptions being thrown, this is to verify we have sane consistency.